src/connectdlg.cpp

```diff
@@ -524,11 +525,18 @@ void CConnectDlg::OnServerAddrEditTextChanged ( const QString& )
 
 void CConnectDlg::OnCustomCentralServerAddrChanged()
 {
-    // only update list if the custom server list is selected
-    if ( pSettings->eCentralServerAddressType == AT_CUSTOM )
+    UpdateDirectoryServerComboBox();
+    if ( pSettings->vstrCentralServerAddress[0] == "" )
     {
-        RequestServerList();
+        pSettings->eCentralServerAddressType = static_cast<ECSAddType> ( AT_DEFAULT );
+        pSettings->iCustomDirectoryIndex     = 0;
     }
+    else
+    {
+        pSettings->eCentralServerAddressType = static_cast<ECSAddType> ( AT_CUSTOM );
+        pSettings->iCustomDirectoryIndex     = 0;
+    }
+    RequestServerList();
```

Comment:
This should only call RequestServerList if the custon directory being changed is the selected one.

Reply:
i agree, but this call to RequestServerList() is from the existing functionaliry.  Volker put it in so that if you change the custom central server on the settings dialog, it updates immediately on the connect dialog.

I am just thinking to leave it alone until we have further discussions on how the custom server management should work.

Reply:
Existing was
```
    // only update list if the custom server list is selected
    if ( pSettings->eCentralServerAddressType == AT_CUSTOM )
    {
        RequestServerList();
    }
```
So no, only if the custom server is the selected _one_ and it's changed (i.e. in settings a different custom server is selected).

We no longer _care_ if the _selected_ custom server _in settings_ changes.  That shouldn't cause a change in the connect dialogue.

We care if the definition of the server selected in _here_ is changed - but that's new and I don't think it's part of this.

Reply:
I don't think so, because the settings dialog puts any modified or selected custom central server at position 0.

Reply:
ah i see

Reply:
I have put back the missing check for AT_CUSTOM and now the functionality is comparable to how it was before - the server list is now not requested if one of the built-in central servers was selected.

Your proposed solution is definitely better - I will try to make it work.

Reply:
You can't avoid re-order if you delete an entry - and if it was the currently selected entry, you're definitely on a different directory server.  So something _will_ be needed here to track what's happening in the client settings dialogue.

Reply:
My view is that if the settings dialog allows re-ordering, then the settings dialog should calculate and return a corrected index.

Right now, if a user changes, on the settings dialog, the list of custom central servers, the connect dialog selects the first custom central server in the list.

I could:
1. leave it as it is
2. reset to the Default built-in central server after any management of the list of custom central servers
3. start modifying the settings dialog to send an index
4. handle it in connect dialog by comparing strings

None is ideal.  But users don't often modify their list of custom central servers.  If they do modify the list, I think option 2 seems like reasonable behaviour.  To me, option 3 seems premature and option 4 seems like it could be a step in the wrong direction.

Reply:
the latest commit implements option 2 above, just so that anyone who is testing can get a feel for what I personally think is the best approach